src/Microsoft.DotNet.CodeAnalysis/Analyzers/PinvokeAnalyzer.cs

```diff
@@ -0,0 +1,157 @@
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(InvalidPinvokeCall); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext obj)
+        {
+            AllowedPinvokeFile = obj.Options.AdditionalFiles.FirstOrDefault(f => Path.GetFileName(f.Path).Contains("PinvokeAnalyzer_"));
+            ExceptionFile = obj.Options.AdditionalFiles.FirstOrDefault(f => Path.GetFileName(f.Path).Contains("PinvokeAnalyzerExceptionList.analyzerdata"));
+            obj.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private Dictionary<string, string> _allowedPinvokes;
+        private Dictionary<string, string> _exceptionPinvokes;
+
+        private Dictionary<string, string> AllowedPinvokes
+        {
+            get
+            {
+                if (_allowedPinvokes == null)
+                {
+                    _allowedPinvokes = ParseAdditionalFile(AllowedPinvokeFile);
+                }
+
+                return _allowedPinvokes;
+            }
+        }
+
+        private Dictionary<string, string> ExceptionPinvokes
+        {
+            get
+            {
+                if (_exceptionPinvokes == null)
+                {
+                    _exceptionPinvokes = ParseAdditionalFile(ExceptionFile);
+                }
+
+                return _exceptionPinvokes;
+            }
+        }
+
+        private AdditionalText ExceptionFile { get; set; }
+        private AdditionalText AllowedPinvokeFile { get; set; }
+
+        private HashSet<string> _isNotSupportedOnWin7 = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private HashSet<string> IsNotSupportedOnWin7 { get { return _isNotSupportedOnWin7; } }
+
+
+        private Dictionary<string, string> ParseAdditionalFile(AdditionalText additionalFile)
+        {
+            Dictionary<string, string> parsedPinvokes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            if (additionalFile != null)
+            {
+                SourceText fileContents = additionalFile.GetText();
+                foreach (TextLine line in fileContents.Lines)
+                {
+                    string lineStr = line.ToString();
+                    if (!string.IsNullOrWhiteSpace(lineStr) && !lineStr.StartsWith("<!--"))
+                    {
+                        string[] splitCount = lineStr.Split('!');
+
+                        if (splitCount.Length == 2 || splitCount.Length == 3)
+                        {
+                            parsedPinvokes[splitCount[1]] = splitCount[0];
+                            if (splitCount.Length == 3)
+                            {
+                                IsNotSupportedOnWin7.Add(splitCount[1]);
```

Comment:
No. SplitCount[2] only contains false to indicate that the method is absent on Win7. The list here is trying to store the methodName and hence is using splitCount[1] instead.